### DISCUSSION TITLE
Quiz

### DISCUSSION THREAD

**Discussion by sebasvelasco353** [sebasvelasco353/FDP-2022_01#137] [General]
Pa Que sirve this?

**Comment by MilaChan7777** [comment]
Camila.
This hace referencia al objeto actual de la clase, lo que se refiere a una instancia concreta de la clase, esta sirve para usar los métodos y atributos de esa clase desde alguno de sus métodos, para llamar a otro de sus constructores o simplemente para pasarle el objeto completo a algún otro método u objeto.

**Comment by Juanaa16** [comment]
Para que sirve utilizar el this en un código? JUANA ALVAREZ

La utilidad principal en java del comando this es para referirse a clases, objetos que están generando un código.
 además de ser la única manera de acceder a un dato, método u objeto, especifica en el código un carácter.

**Comment by Kazura-Collins** [comment]
En general, el valor de this está determinado por cómo se invoca a la función. No puede ser establecida mediante una asignación en tiempo de ejecución, y puede ser diferente cada vez que la función es invocada.

**Comment by Kathereyes19** [comment]
Katherine Reyes 

respuesta: 
This es un atajo que lo que hace es decirle a un objeto desde donde se invoca la funcion. Es decir,  que con this podemos acceder a todos los valores y propiedades del objeto origen. Cualquier cambio realizado a this también actualizará el objeto referido.
this depende de diferentes factores:
1. el objeto desde donde se invocó la función.
3. si el alcance está en modo estricto/no estricto.

**Comment by Juli0931** [comment]
Juliana Buitrago
.this es un comando utilizado en javaScript para hacer referencia a diferentes objetos dependiendo de como se use. 
Suele ser utilizado principalmente en los métodos para referirse al objeto como tal.

**Comment by ghost** [comment]
Sofía Motta
.This: es un keyword de JavaScript que tiene un comportamiento muy diferente a otros lenguajes de programación, así para algunos es considerado uno de los grandes errores de diseño del lenguaje.

Para entender mas el THIS hace a referencia al objetos, que contiene el método donde se invoca dentro de un método.

**Comment by AnnaHenao131** [comment]
Ana Henao: Sirve para referirse a clases, objetos que está generando un código, además de ser la única manera de acceder a un dato, método u objeto, específico en el código. Jiji :3

**Comment by Santygr21** [comment]
Santiago González.
This hace referencia al objeto global, determinado por su llamado en la función. Este se usa principalmente en métodos para hacer referencia a un objeto.

**Comment by DilanStivenCano** [comment]
Dilan Cano
El this hace referencia a la instancia actual de la clase y también se usa como modificador del primer parámetro de un método de extensión.

**Comment by Karolfc** [comment]
En JavaScript, this se usa para apuntar o señalar el objeto que está llamando a la función. La función de this depende del lugar en el que se esté empleando dentro del código.

**Comment by Darksea1220** [comment]
Douglas Osorio
.this  funciona para indicarle al programa (js) que te estás refiriendo a un objeto previamente creado, dentro del mismo objeto, en otras palabras, con .this le dices a un objeto que el parámetro a continuación es para sí mismo. :v

**Comment by Pipelenisf** [comment]
El `.this` en JS es una keyword que se usa para referirse a un objeto.

**Comment by IBaronap** [comment]
Isabella Barona
This es un comando usado en javascript para referirse a un objeto. Su valor cambia según como este siendo usado, en donde se encuentra en la función.
